zulipterminal/ui_tools/views.py

```diff
@@ -1441,3 +1441,97 @@ def keypress(self, size: urwid_Size, key: str) -> str:
             )
             return key
         return super().keypress(size, key)
+
+
+class FullMsgView(urwid.Frame):
+    def __init__(self, controller: Any, message: Message,
+                 message_links: 'OrderedDict[str, Tuple[str, int, bool]]',
+                 time_mentions: List[Tuple[str, str]],
+                 title: str) -> None:
+        self.title = title
+        self.controller = controller
+        self.message = message
+        self.message_links = message_links
+        self.time_mentions = time_mentions
+
+        msg_box = MessageBox(message, controller.model, None)
+        if not msg_box.need_recipient_header():
+            msg_box.widget_list.insert(0, msg_box.top_header_bar(msg_box))
+
+        # Setting message header
+        header = msg_box.widget_list[:2]
+        header = header + [urwid.Divider()]
+
+        # Fetch and store raw and formatted message
+        _content_formatted = msg_box.widget_list[2].original_widget[-1]
+        response = controller.model.client.get_raw_message(message['id'])
+        _content_raw = urwid.Text('N/A')
+        if response['result'] == 'success':
+            _content_raw = urwid.Columns([
+                (1, urwid.Text('')),
+                urwid.Text(response['raw_content']),
+            ])
+
+        tab_list = [
+            {'label': 'Formatted content', 'content': _content_formatted},
+            {'label': 'Raw content',       'content': _content_raw},
+        ]
+
+        self.tab_labels = [tab['label'] for tab in tab_list]
+        self.tab_contents = [tab['content'] for tab in tab_list]
+        self.tab_focus = 0  # Initial tab
+
+        self._render_body()  # Responsible render the body
+
+        super().__init__(self.contents['body'][0], (urwid.Pile(header, None)),
+                         focus_part='body')
+
+        max_cols, max_rows = controller.maximum_popup_dimensions()
+
+        self.width = min(max_cols, 64)
+
+        height = self.calculate_popup_height(
+            (self.contents['body'][0].body
+             + self.contents['header'][0].widget_list),
+            self.width
+        )
+        self.height = min(max_rows, height)
```

Comment:
True. It would be then good to first show the raw content then? (Since mostly the length of raw content is more than the formatted one).

On a side note, I'm a bit skeptical on whether to change the width/height of the popup upon switching. Is this something we want to achieve?